### PR TITLE
pdctl: omit state which is meanless in StoreInfo

### LINKED ISSUE #3816: state should not be printed which is meanless
### What did you do?

Use pd-ctl to show store info.

### What did you expect to see?

```
{
  "store": {
    "id": 3,
    "address": "127.0.0.1:20161",
    "version": "5.1.0-alpha",
    "status_address": "127.0.0.1:20181",
    "git_hash": "45ce5b9584d618bc777877be8a77cb94f61b8410",
    "start_timestamp": 1624616146,
    "deploy_path": "/home/ryan/.tiup/components/tikv/v5.0.0-nightly-20210624",
    "last_heartbeat": 1624616236412333050,
    "state_name": "Offline"
  },
  "status": {
    "capacity": "271.9GiB",
    "available": "53.23GiB",
    "used_size": "31.59MiB",
    "leader_count": 5,
    "leader_weight": 1,
    "leader_score": 5,
    "leader_size": 5,
    "region_count": 25,
    "region_weight": 1,
    "region_score": 211220.3015687649,
    "region_size": 25,
    "start_ts": "2021-06-25T18:15:46+08:00",
    "last_heartbeat_ts": "2021-06-25T18:17:16.41233305+08:00",
    "uptime": "1m30.41233305s"
  }
}
```

### What did you see instead?
```
{
  "store": {
    "id": 3,
    "address": "127.0.0.1:20161",
    "state": 1,
    "version": "5.1.0-alpha",
    "status_address": "127.0.0.1:20181",
    "git_hash": "45ce5b9584d618bc777877be8a77cb94f61b8410",
    "start_timestamp": 1624616146,
    "deploy_path": "/home/ryan/.tiup/components/tikv/v5.0.0-nightly-20210624",
    "last_heartbeat": 1624616236412333050,
    "state_name": "Offline"
  },
  "status": {
    "capacity": "271.9GiB",
    "available": "53.23GiB",
    "used_size": "31.59MiB",
    "leader_count": 5,
    "leader_weight": 1,
    "leader_score": 5,
    "leader_size": 5,
    "region_count": 25,
    "region_weight": 1,
    "region_score": 211220.3015687649,
    "region_size": 25,
    "start_ts": "2021-06-25T18:15:46+08:00",
    "last_heartbeat_ts": "2021-06-25T18:17:16.41233305+08:00",
    "uptime": "1m30.41233305s"
  }
}
```

`state` has no meaning here. We should remove it in output.

### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?
fix #3816 

### What is changed and how it works?
In pdctl, change MetaStore.State value into 0 to omit state in JSON output.

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/pdctl/helper.go [hunk 1]
```diff
@@ -53,6 +54,7 @@ func CheckStoresInfo(c *check.C, stores []*api.StoreInfo, want []*metapb.Store)
 	}
 	for _, s := range stores {
 		obtained := proto.Clone(s.Store.Store).(*metapb.Store)
+		obtained.State = metapb.StoreState(metapb.StoreState_value[s.Store.StateName])
```
Comment (rleungx) [inline tests/pdctl/helper.go:57]: `StateName` contains `Down` and `Disconnected` which are not contained in `StoreState_value`.

File: tools/pd-ctl/pdctl/command/global.go [hunk 2]
```diff
@@ -57,6 +57,8 @@ func InitHTTPSClient(caPath, certPath, keyPath string) error {
 	return nil
 }
 
+type convertOutputFunc func(string) string
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/global.go:60]: I think there is no need to define `convertOutputFunc`.

Comment (CabinfeverB) [reply tools/pd-ctl/pdctl/command/global.go:60]: I see. We can use `:=` for declaration

File: tests/pdctl/helper.go (hunk 1 above)
Comment (CabinfeverB) [reply tests/pdctl/helper.go:57]: I think it's a way to declare a new map for  `StoreState_value`

File: tests/pdctl/helper.go [hunk 3]
```diff
@@ -33,6 +33,15 @@ import (
 	"github.com/tikv/pd/tests"
 )
 
+// StoreState_value is used for store state in PD definition
+var StoreState_value = map[string]int32{
+	"Up":           0,
+	"Disconnected": 0,
```
Comment (rleungx) [inline tests/pdctl/helper.go:39]: `Disconnected` could be `Offline` at the same time.

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 4]
```diff
@@ -243,18 +244,50 @@ func storeLimitSceneCommandFunc(cmd *cobra.Command, args []string) {
...
 		return
 	}
+	cFunc := convertToStoresInfo
 	if len(args) == 1 {
 		if _, err := strconv.Atoi(args[0]); err != nil {
 			cmd.Println("store_id should be a number")
 			return
 		}
 		prefix = fmt.Sprintf(storePrefix, args[0])
+		cFunc = convertToStoreInfo
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:290]: Does `else` also need it?

Comment (CabinfeverB) [reply tools/pd-ctl/pdctl/command/store_command.go:290]: Like `prefix`, `cFunc` has assigned before `if` statement

File: tests/pdctl/helper.go (hunk 3 above)
Comment (CabinfeverB) [reply tests/pdctl/helper.go]: has romoved it

**Review by rleungx — APPROVED** [review]

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 5]
```diff
@@ -243,18 +244,50 @@ func storeLimitSceneCommandFunc(cmd *cobra.Command, args []string) {
...
+	return string(byteArr)
+}
+
+func convertToStoresInfo(content string) string {
+	stores := &api.StoresInfo{}
+	err := json.Unmarshal([]byte(content), stores)
+	if err != nil {
+		return content
+	}
+	for _, store := range stores.Stores {
```
Comment (JmPotato) [inline tools/pd-ctl/pdctl/command/store_command.go:267]: What about adding a comment here to explain why we change this to 0?

Comment (CabinfeverB) [reply tools/pd-ctl/pdctl/command/store_command.go:271]: done

**Review by JmPotato — APPROVED** [review]

**Pull request merged**